iac-extensions/docker/src/test/resources/checks/ArgDefinedOutsideOfScopeCheck/Dockerfile

```diff
@@ -0,0 +1,38 @@
+ARG BUSYBOX_TAG="latest"
+ARG SETTINGS
+ARG SETTINGS_2="--default-settings"
+ARG USER=builder
+
+FROM busybox:$BUSYBOX_TAG
+RUN ./run/setup
+
+FROM busybox
+# Noncompliant@+1 {{Include the ARG instruction in the build stage where it is used.}}
+RUN ./run/setup $SETTINGS
+#               ^^^^^^^^^
+
+FROM busybox
+ARG SETTINGS
+RUN ./run/setup $SETTINGS
+
```

Comment:
You could add here another occurence of this:
```
FROM busybox
# Noncompliant@+1
RUN ./run/setup $SETTINGS
```
It would show that the algorithm is working for this rule, as the `Symbol` table is not mixing this variable symbol in `RUN` with the one redeclared at line 15 in `ARG`.
I would add a comment about this also.